### DISCUSSION TITLE
Display company logo + product name in header

### DISCUSSION THREAD

**Discussion by wickham-c** [filamentphp/filament#11918] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.50

### How can we help you?

What I'm attempting to achieve is to include a company logo and a product name (alongside the logo) in the header of a Filament panel.

Unfortunately, it seems I cannot use both the `brandName` and `brandLogo` features to achieve this as they are employed in a mutually exclusive manner.

I have also attempted to override the `logo.blade.php` template but, perhaps due to my lack of CSS knowledge, I'm unable to get the product name text to appear to the right of the company logo. 

Ideally, what I'd like to do is add a new `<div>` element alongside (i.e. as a sibling to) the `<div>` element that already houses the logo. I can do this by overriding the `/sidebar/index.blade.php` but this feels "dirty" as that template is quite a significant one.

In your opinion, which of the following would be the best way forward...

1. Override the `/sidebar/index.blade.php` file?
2. Override the `logo.blade.php` file (and tweak the layout via CSS)?
3. Use some other approach?

Any input would be much appreciated.

**Comment by ArtMin96** [comment, marked as answer]
[Adding a logo](https://filamentphp.com/docs/3.x/panels/themes#adding-a-logo)
```php
use Filament\Panel;
 
public function panel(Panel $panel): Panel
{
    return $panel
        // ...
        ->brandLogo(fn () => view('filament.admin.logo'));
}
```

```html
<div class="flex items-center space-x-4">
    <svg
        viewBox="0 0 128 26"
        xmlns="http://www.w3.org/2000/svg"
        class="h-full fill-gray-500 dark:fill-gray-400"
    >
        <!-- ... -->
    </svg>

    Brand name
</div>
```

**Reply by wickham-c** [reply]
Many thanks @ArtMin96 - I was missing the `flex` class on the `div` container.

Just out of interest, if you also supply a dark-mode version of the logo blade template (which contains SVG elements which are filled with white), does the dark-mode version seem to permanently override the light-mode version i.e. even when the user is in light-mode, or have I misconfigured something?  I'm seeing a white logo on a white background in light-mode, and the same white logo on a dark background in dark-mode, yet the SVG styles are set to contrast the background. If I comment out the `darkModeBrandLogo` feature then I see the expected colours from the light-mode logo.
```php
class AppPanelProvider extends PanelProvider
{
    public function panel(Panel $panel): Panel
    {
        return $panel
            // ...
            ->brandLogo(fn () => view('filament.admin.logo'))
            ->darkModeBrandLogo(fn () => view('filament.admin.logo-darkmode'))
            // ...
```

**Reply by arbazajaz** [reply]
good work
